core/src/main/java/org/elasticsearch/index/get/GetResult.java

```diff
@@ -247,23 +245,71 @@ public XContentBuilder toXContentEmbedded(XContentBuilder builder, Params params
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
         if (!isExists()) {
-            builder.field(Fields._INDEX, index);
-            builder.field(Fields._TYPE, type);
-            builder.field(Fields._ID, id);
-            builder.field(Fields.FOUND, false);
+            builder.field(_INDEX, index);
+            builder.field(_TYPE, type);
+            builder.field(_ID, id);
+            builder.field(FOUND, false);
         } else {
-            builder.field(Fields._INDEX, index);
-            builder.field(Fields._TYPE, type);
-            builder.field(Fields._ID, id);
+            builder.field(_INDEX, index);
+            builder.field(_TYPE, type);
+            builder.field(_ID, id);
             if (version != -1) {
-                builder.field(Fields._VERSION, version);
+                builder.field(_VERSION, version);
             }
             toXContentEmbedded(builder, params);
         }
+        builder.endObject();
         return builder;
     }
 
+    public static GetResult fromXContent(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.nextToken();
+        assert token == XContentParser.Token.START_OBJECT;
```

Comment:
leftover, I never know whether I want to use asserts or exceptions. I also don't want to end up validating that our own response are well formed, cause we should have tests for that, but I guess these few situations need to be checked. The question is exceptions or asserts?